File: solutions/cpp/01-ry8/code/src/main.cpp
#include <fstream>
#include <iostream>
#include <sstream>
#include <string>

int main(int argc, char *argv[]) {
  // Flush after every std::cout / std::cerr
  std::cout << std::unitbuf;
  std::cerr << std::unitbuf;

  if (argc < 3) {
    std::cerr << "Usage: ./your_program.sh tokenize <filename>" << std::endl;
    return 1;
  }

  std::string command{argv[1]};
  std::string filename{argv[2]};

  if (command != "tokenize") {
    std::cerr << "Unknown command: " << command << std::endl;
    return 1;
  }

  std::ifstream file{filename};
  using Iterator = std::istreambuf_iterator<char>;
  std::string file_contents{Iterator{file}, Iterator{}};

  if (!file_contents.empty()) {
    std::cerr << "Scanner not implemented" << std::endl;
  } else {
    std::cout << "EOF  null" << std::endl; // Placeholder, remove this line when implementing the scanner
  }
}

File: solutions/ruby/01-ry8/code/app/main.rb
command, filename = ARGV
unless filename
  abort "Usage: ./your_program.sh tokenize <filename>"
end

abort "Unknown command: #{command}" if command != "tokenize"

file_contents = begin
  File.read(filename)
rescue SystemCallError
  ""
end

unless file_contents.empty?
  raise "Scanner not implemented"
else
  puts "EOF  null" # Placeholder, replace this line when implementing the scanner
end
